src/cljs/main/broadfcui/common.cljs

```diff
@@ -248,3 +248,6 @@
 
 (defn has-terra-return? []
   (string/includes? js/window.location.search "return=terra"))
+
+(defn has-firecloud-return? []
+  (string/includes? js/window.location.search "return=firecloud"))
```

Comment:
I suggest adding a helper function here like:
```
(defn has-return? []
  (or (has-terra-return?) (has-firecloud-return?)))
```
and using this helper everywhere else you have ` (or (has-terra-return?) (has-firecloud-return?))`, to keep all the other code clean